numpy/core/tests/test_getlimits.py

```diff
@@ -86,7 +87,15 @@ def test_finfo_repr(self):
 
 def test_instances():
     iinfo(10)
+    iinfo(np.int16(1))
     finfo(3.0)
+    finfo(np.float32(1.))
+    finfo(np.float64(-1.))
```

Comment:
I suppose these are just meant to make sure there's no error/crash on these calls?

I wonder if we'd gain slightly more with a property check while we're at it, i.e., something like:

```python
# check instances and correct max attributes
assert finfo(np.float64(-1.)).max > finfo(np.float32(1.)).max
```

Reply:
@tylerjereddy I am waiting on the outcome of discussions at #23108 and #23088 on whether we should require that
`finfo(np.float32(1.)) is  finfo(np.float32)` (or only `finfo(np.float32(1.)) ==  finfo(np.float32)`.

Based on that I will indeed start testing a bit more.